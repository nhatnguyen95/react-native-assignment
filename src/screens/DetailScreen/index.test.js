import "react-native";
import React from "react";
import { DetailScreen } from "./index";
import renderer from "react-test-renderer";

describe("DetailScreen", () => {
  it("should match to snapshot", () => {
    const wrapper = renderer.create(<DetailScreen />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should set trip index to state after navigating", () => {
    const route = {
      params: { tripIndex: 10 },
    };
    wrapper = renderer.create(<DetailScreen route={route} />);
    expect(wrapper.getInstance().state.tripIndex).toBe(
      route.params.tripIndex
    );
  });
});
