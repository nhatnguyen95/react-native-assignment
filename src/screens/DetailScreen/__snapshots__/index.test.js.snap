// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <RCTScrollView
    testID="scroll-view"
  >
    <View />
  </RCTScrollView>
</View>
`;
